hydra-explorer/src/Hydra/Explorer/ExplorerState.hs

```diff
@@ -262,31 +279,74 @@ aggregateFanoutObservation headId explorerState =
  where
-  aggregateOnChainTx :: ExplorerState -> OnChainTx Tx -> ExplorerState
-  aggregateOnChainTx explorerState =
+  aggregateOnChainTx :: ExplorerState -> HeadObservationAt -> ExplorerState
+  aggregateOnChainTx ExplorerState{heads} =
     \case
-      OnInitTx{headId, headSeed, headParameters, participants} -> aggregateInitObservation headId headSeed headParameters participants explorerState
-      OnAbortTx{headId} -> aggregateAbortObservation headId explorerState
-      OnCommitTx{headId, party, committed} -> aggregateCommitObservation headId party committed explorerState
-      OnCollectComTx{headId} -> aggregateCollectComObservation headId explorerState
-      OnCloseTx{headId, snapshotNumber, contestationDeadline} -> aggregateCloseObservation headId snapshotNumber contestationDeadline explorerState
-      OnContestTx{headId, snapshotNumber} -> aggregateContestObservation headId snapshotNumber explorerState
-      OnFanoutTx{headId} -> aggregateFanoutObservation headId explorerState
+      HeadObservationAt{point, blockNo, onChainTx = Just OnInitTx{headId, headSeed, headParameters, participants}} ->
+        ExplorerState
+          { heads = aggregateInitObservation headId point blockNo headSeed headParameters participants heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnAbortTx{headId}} ->
+        ExplorerState
+          { heads = aggregateAbortObservation headId point blockNo heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnCommitTx{headId, party, committed}} ->
+        ExplorerState
+          { heads = aggregateCommitObservation headId point blockNo party committed heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnCollectComTx{headId}} ->
+        ExplorerState
+          { heads = aggregateCollectComObservation headId point blockNo heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnCloseTx{headId, snapshotNumber, contestationDeadline}} ->
+        ExplorerState
+          { heads = aggregateCloseObservation headId point blockNo snapshotNumber contestationDeadline heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnContestTx{headId, snapshotNumber}} ->
+        ExplorerState
+          { heads = aggregateContestObservation headId point blockNo snapshotNumber heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Just OnFanoutTx{headId}} ->
+        ExplorerState
+          { heads = aggregateFanoutObservation headId point blockNo heads
+          , point
+          , blockNo
+          }
+      HeadObservationAt{point, blockNo, onChainTx = Nothing} ->
+        ExplorerState
+          { heads
+          , point
+          , blockNo
```

Comment:
Must: display these in the user interface

I don't see where the last seen point/block number is shown in the user interface? I would expect the API yaml to contain it somewhere and the user interface to show it outside of the table as these data points are not related to some specific head.